packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,710 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as logs from '@aws-cdk/aws-logs';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+import * as awscli from '@aws-cdk/lambda-layer-awscli';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * The props for a EMR Containers StartJobRun Task.
+ */
+export interface EmrContainersStartJobRunProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * The virtual cluster ID for which the job run request is submitted.
+   */
+  readonly virtualClusterId: sfn.TaskInput;
+
+  /**
+   * The name of the job run.
+   *
+   * @default - No job name
+   */
+  readonly jobName?: string;
+
+  /**
+   * The execution role for the job run.
+   *
+   * @default - Automatically generated
+   */
+  readonly executionRole?: iam.IRole;
+
+  /**
+   * The Amazon EMR release version to use for the job run.
+   *
+   * @example - ReleaseLabel.EMR_6_3_0
+   */
+  readonly releaseLabel: ReleaseLabel;
+
+  /**
+   * The configurations for the application running by the job run.
+   * Maximum of 100 items
+   *
+   * @see https://docs.aws.amazon.com/emr-on-eks/latest/APIReference/API_Configuration.html
+   *
+   * @default - No application config
+   */
+  readonly applicationConfig?: ApplicationConfiguration[];
+
+  /**
+   * The job driver for the job run.
+   *
+   * @see https://docs.aws.amazon.com/emr-on-eks/latest/APIReference/API_JobDriver.html
+   *
+   * @default - No job driver
+   */
+  readonly jobDriver: JobDriver;
```

Comment:
Sorry looks like I made a mistake that caused some confusion, the reason why a Job Driver should be required is that it contains the Job (PySpark script, etc) that's being run. If there's no job, the job can't be run.

Reply:
ah okay. I was confused because of the mismatch between the docs and this prop being required, as well as the API I believe also having it optional (which doesn't make sense given your explanation)